server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -151,6 +154,7 @@ public Response doPost(
         );
       }
 
+      Thread.currentThread().setName("qtp-query-" + queryId);
```

Comment:
SLF4J's MDC should be compatible with the Log4j one. If MDC is elected here, I suggest using that one.